--- crt.pp.O	2014-01-03 14:19:02.000000000 -0800
+++ crt.pp	2016-01-31 19:24:56.684336452 -0800
@@ -17,6 +17,9 @@
 
 {$i crth.inc}
 
+Procedure ShiftAltCharset;
+Procedure ShiftNormalCharset;
+
 Const
   { Controlling consts }
   Flushing     = false;               {if true then don't buffer output}
@@ -1451,15 +1454,25 @@
 
 procedure CursorOn;
 begin
-  ttySendStr(#27'[?2c');
+  ttySendStr(#27'[?25h');
 end;
 
 
 procedure CursorOff;
 begin
-  ttySendStr(#27'[?1c');
+  ttySendStr(#27'[?25l');
+end;
+
+
+Procedure ShiftAltCharset;
+begin
+  ttySendChar(#14);
 end;
 
+Procedure ShiftNormalCharset;
+begin
+  ttySendChar(#15);
+end;
 
 {******************************************************************************
                                Initialization
@@ -1634,8 +1647,9 @@
         CurrY:=1;
         ttySendStr(#27'[H');
       end;
-   {Reset Attribute (TextAttr=7 at startup)}
-      ttySendStr(#27'[m');
+   {Reset Attribute (TextAttr=7 at startup)
+    and enable ACS character set}
+      ttySendStr(#27'[m'#27')0');
     end;
 
 Finalization
