--- crt.pp.O	2005-02-14 09:13:31.000000000 -0800
+++ crt.pp	2005-04-02 01:03:20.000000000 -0800
@@ -18,6 +18,9 @@
 
 {$i crth.inc}
 
+Procedure ShiftAltCharset;
+Procedure ShiftNormalCharset;
+
 Const
   { Controlling consts }
   Flushing     = false;               {if true then don't buffer output}
@@ -1432,15 +1435,25 @@
 
 procedure CursorOn;
 begin
-  ttySendStr(#27'[?2c');
+  ttySendStr(#27'[?25h');
 end;
 
 
 procedure CursorOff;
 begin
-  ttySendStr(#27'[?1c');
+  ttySendStr(#27'[?25l');
+end;
+
+
+Procedure ShiftAltCharset;
+begin
+  ttySendChar(#14);
 end;
 
+Procedure ShiftNormalCharset;
+begin
+  ttySendChar(#15);
+end;
 
 {******************************************************************************
                                Initialization
@@ -1607,8 +1620,9 @@
         CurrY:=1;
         ttySendStr(#27'[H');
       end;
-   {Reset Attribute (TextAttr=7 at startup)}
-      ttySendStr(#27'[m');
+   {Reset Attribute (TextAttr=7 at startup)
+    and enable ACS character set}
+      ttySendStr(#27'[m'#27')0');
     end;
 
 Finalization
